crates/transaction-pool/src/pool/txpool.rs

```diff
@@ -546,6 +573,52 @@ impl<T: TransactionOrdering> TxPool<T> {
         self.all_transactions.txs_iter(sender).map(|(_, tx)| Arc::clone(&tx.transaction)).collect()
     }
 
+    /// Updates only the pending fees without triggering subpool updates.
+    /// Returns the previous base fee and blob fee values.
+    #[allow(clippy::missing_const_for_fn)]
+    fn update_pending_fees_only(
+        &mut self,
+        mut new_base_fee: u64,
+        new_blob_fee: Option<u128>,
+    ) -> (u64, u128) {
+        std::mem::swap(&mut self.all_transactions.pending_fees.base_fee, &mut new_base_fee);
+
+        let prev_blob_fee = if let Some(mut blob_fee) = new_blob_fee {
+            std::mem::swap(&mut self.all_transactions.pending_fees.blob_fee, &mut blob_fee);
+            blob_fee
+        } else {
+            self.all_transactions.pending_fees.blob_fee
+        };
+
+        (new_base_fee, prev_blob_fee)
+    }
+
+    /// Applies fee-based promotion updates.
+    /// Directly modifies the provided outcome by pushing promoted transactions.
+    fn apply_fee_updates(
+        &mut self,
+        prev_base_fee: u64,
+        prev_blob_fee: u128,
+        outcome: &mut UpdateOutcome<T::Transaction>,
+    ) {
+        let current_base_fee = self.all_transactions.pending_fees.base_fee;
+        let current_blob_fee = self.all_transactions.pending_fees.blob_fee;
+
+        let base_fee_decreased = current_base_fee < prev_base_fee;
+        let blob_fee_decreased = current_blob_fee < prev_blob_fee;
+
+        // Only handle promotions when base fee decreases
+        if base_fee_decreased {
+            self.handle_basefee_decrease(current_base_fee, |tx| outcome.promoted.push(tx.clone()));
+            // Blob promotions depend on both base- and blob-fee affordability, so recheck blob
+            // transactions as well when the base fee drops.
+            self.handle_blob_fee_decrease(|tx| outcome.promoted.push(tx.clone()));
```

Comment:
looking at 
```
  (Ordering::Less, _) | (_, Ordering::Less) => {
      // decreased blob/base fee: recheck blob
  pool and promote all that are now valid
```

I am assuming that blob transactions can be promoted when either the blob fee OR the base fee decreases.

This makes sense because blob transactions need to satisfy both base fee and blob fee requirements.